js-pkg/packages/driftdb-react/src/index.tsx

```diff
@@ -108,11 +108,16 @@ export function useSharedState<T>(key: string, initialValue: T): [T, SetterFunct
   const db = useDatabase()
   const [state, setInnerState] = React.useState<T>(initialValue)
 
-  const stateListener = useRef<StateListener<T>>(null)
+  const stateListener = useRef<StateListener<SetStateAction<T>> | null>(null)
+
   if (stateListener.current === null) {
-    ;(stateListener as any).current = new StateListener(setInnerState, db, key)
+    stateListener.current = new StateListener(setInnerState, db, key)
   }
 
+  useEffect(() => {
+    stateListener.current!.subscribe()
+  }, [stateListener.current])
```

Comment:
This needs to return a teardown fn that unsubscribes from the db updates when the component is removed.